cpp/CMakeLists.txt

```diff
@@ -55,19 +55,14 @@ endfunction(cpp_cc_disable_precommit)
 
 function(cpp_cc_enable_clang_formatting)
   find_package(PythonInterp 3.5 REQUIRED)
-  find_package(ClangFormat REQUIRED)
+  if(NOT ${CODING_CONV_PREFIX}_ClangFormat_REQUIRED_VERSION)
+    set(${CODING_CONV_PREFIX}_ClangFormat_REQUIRED_VERSION 7)
+  endif()
+  find_package(ClangFormat ${${CODING_CONV_PREFIX}_ClangFormat_REQUIRED_VERSION} REQUIRED)
   if(NOT ClangFormat_FOUND)
     message(SEND_ERROR "Could not find Clang-Format")
     return()
   endif()
-  if(ClangFormat_VERSION_MAJOR GREATER 6 AND ClangFormat_VERSION_MAJOR LESS 14)
-    set(${CODING_CONV_PREFIX}_ClangFormat_CONFIG_FILE clang-format-${ClangFormat_VERSION_MAJOR})
```

Comment:
I'm wondering what happens with the configuration file now? A quick `git grep` says we still read from `..._ClangFormat_CONFIG_FILE` but that we don't set it any more.

I think the desired behaviour is that we use the newest `clang-format-X` file in this repo that is not tagged for a newer version of the `clang-format` executable than we find.
e.g. if the repo contains `clang-format-11` and `clang-format-12` config files then:
- `clang-format-11` executable should use the `clang-format-11` config
- `clang-format-12` executable should use the `clang-format-12` config
- `clang-format-13` executable should use the `clang-format-12` config

Reply:
The 2 last commits should fulfill your expectations @olupton  :)